### ISSUE TITLE
False positive PLC0132 for NewType wrapped in Annotated

### ISSUE THREAD

**Issue by minmax** [astral-sh/ruff#10840]
Ruff 0.3.5 false positive PLC0132 for NewType wrapped in Annotated

```python
MyIds = Annotated[NewType("WrongName", str), int] # PLC0132 doesn't work

MyId = NewType("Id", str) # NewType` name `Id` does not match assigned variable name `MyId` PLC0132
```

Keywords: Annotated, NewType, PLC0132

**Comment by charliermarsh** [comment]
Thanks!

**Comment by minmax** [comment]
@erictraut [say's](https://github.com/microsoft/pyright/issues/7645#issuecomment-2043962935) that it is not a bug, so maybe this issuer need to be closed without fix/code.
 
> Pyright is working correctly here. When you wrap a `NewType` with `Annotated`, the resulting object is no longer a `NewType`. It's a type alias. It's name does not need to match that of the (anonymous) NewType that was created.


Related task in pyright repo: [7645](https://github.com/microsoft/pyright/issues/7645)

**Comment by charliermarsh** [comment]
Ah ok, thanks for following up!
